File: widgetssdk/src/main/java/com/glia/widgets/call/CallViewCallback.java
package com.glia.widgets.call;

import androidx.annotation.NonNull;

import com.glia.androidsdk.engagement.Survey;

public interface CallViewCallback {

    void emitState(CallState callState);

    void navigateToChat();

    void navigateToSurvey(@NonNull Survey survey);

    void destroyView();
}

File: widgetssdk/src/main/java/com/glia/widgets/call/CallViewCallback.kt
package com.glia.widgets.call

import com.glia.androidsdk.engagement.Survey

internal interface CallViewCallback {
    fun emitState(callState: CallState)
    fun navigateToChat()
    fun navigateToSurvey(survey: Survey)
    fun destroyView()
    fun minimizeView()
    fun showMissingPermissionsDialog()
    fun showEngagementConfirmationDialog()
    fun navigateToWebBrowserActivity(title: String, url: String)
}
